jina/executors/indexers/vector/ngt.py

```diff
@@ -0,0 +1,65 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Tuple
+
+import numpy as np
+from .numpy import NumpyIndexer
+
+
+class NGTIndexer(NumpyIndexer):
+    """NGT powered vector indexer
+
+    For more information about the NGT supported parameters and installation problems, please consult:
+        - https://github.com/yahoojapan/NGT
+
+    .. note::
+        NGT package dependency is only required at the query time.
+        Quick Install : pip install ngt
+    """
+
+    def __init__(self, num_threads: int = 2, metric: str = 'L2', epsilon: int = 0.1, *args, **kwargs):
+        """
+        Initialize an NGT Indexer
+        :param num_threads: Number of threads to build index
+        :param metric: Should be one of {L1,L2,Hamming,Jaccard,Angle,Normalized Angle,Cosine,Normalized Cosine}
+        :param epsilon: Toggle this variable for speed vs recall tradeoff.
+                        Higher value of epsilon means higher recall
+                        but query time will increase with epsilon
+        """
+
+        super().__init__(*args, **kwargs)
+        self.metric = metric
+        self.index_path = 'index'
```

Comment:
@joaopalotti the index_path can be anything really. It doesn't get write/add event. We just use it to declare a NGT index variable because its a required parameter. We aren't actually gonna use it.

Reply:
is it possible to set it to None then? Meaning ngtpy.Index(None)

Reply:
Is it possibe to set it to None then?

Reply:
Hey @JoanFM ! When a NGT Index object is declared a directory is created with the index_path parameter. So setting it to None will raise an error. Therefore I'm using just any index path other than None.

Reply:
We won't be saving anything in the directory till we call index.save() so we're good !

Reply:
Thank you for your quick attention to the comments, sorry not familiar with the library.